includes/wikia/GlobalFunctions.php

```diff
@@ -1557,3 +1557,28 @@ function wfGetBaseDomainForHost( $host ) {
 
 	return $wgWikiaBaseDomain;
 }
+
+/**
+ * Returns true for URLs with fandom domain, some examples:
+ * - https://starwars.fandom.com/wiki/Yoda
+ * - starwars.fandom.com/wiki/Yoda
+ * - starwars.fandom-dev.pl/
+ * - starwars.fandom-dev.us
+ *
+ * @param $url
+ * @return bool
+ */
+function wfDoesURLHaveFandomDomain( $url ): bool {
```

Comment:
One thought here, what do you think of renaming this method to something like `wfForceHttpsForURL` or something similar? My thinking is this is the intent of the method, and decouples it from a specific domain/implementation.

Reply:
@Grunny I think you misclicked or misspelled, this method is not about forcing HTTPS at all.

Reply:
This method itself isn't, but everywhere it's used is, right? It's used in place of `wgEnableHTTPSForAnons`. What I'm thinking is that this method should be decoupled from the fandom.com domain implementation, so if we enabled HTTPS by default on any other domains in the future, we don't need to change all these method calls, but instead just add the domains to the implementation of this method with a name like `wfHttpsEnabledForURL`. There may be a use for `wfDoesURLHaveFandomDomain` separately in the future, which could be used by `wfHttpsEnabledForURL`, but right now we only need to know if HTTPS is enabled by default for the URL. What do you think?

Reply:
I see. It'll be similar to `wfHttpsAllowedForURL` but I think it makes sense :+1: Updated in https://github.com/Wikia/app/pull/16076/commits/39269b4aec40657e0907b12a01d39871d75edffb